### PR TITLE
Saas 12.2 elearning various fixes qmo

### PR DESCRIPTION
**Pull request by qmo-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_slides/views/website_slides_templates_lesson.xml [hunk 1]
```diff
@@ -318,17 +318,19 @@
...
+                            <span class="d-flex pl-3">
+                                <span class="badge badge-pill" t-esc="slide.likes"/>
+                                <span class="mr-5 ml-auto">Likes</span>
+                            </span>
+                            <span class="d-flex pl-3">
+                                <span class="badge badge-pill" t-esc="slide.dislikes"/>
+                                <span class="mr-5 ml-auto">Dislikes</span>
+                            </span>
+                        </t>
+                        <span t-if="slide.channel_id.allow_comment" class="d-flex pl-3">
```
Comment (dbeguin) [inline addons/website_slides/views/website_slides_templates_lesson.xml]: I think you can only filter on slide.channel_id.allow_comment in the parent div. and filter again on
slide.channel_id.channel_type == 'documentation' like it's already done. + remove this t-if here.

File: addons/website_slides/views/website_slides_templates_lesson.xml [hunk 2]
```diff
@@ -191,7 +191,7 @@
                     t-att-href="'/slides/slide/%s' % (slug(previous_slide)) if previous_slide else '#'">
                     <i class="fa fa-chevron-left mr-2"></i> Prev
                 </a>
-                <t t-set="allow_done_btn" t-value="slide.slide_type in ['infographic', 'presentation', 'document', 'webpage'] and not slide.question_ids and not channel_progress[slide.id].get('completed')"/>
+                <t t-set="allow_done_btn" t-value="slide.slide_type  in ['infographic', 'presentation', 'document', 'webpage', 'video'] and not slide.question_ids and not channel_progress[slide.id].get('completed')"/>
```
Comment (dbeguin) [inline addons/website_slides/views/website_slides_templates_lesson.xml]: remove space before 'in'

**Comment by dbeguin** [comment]
Can you apply commit message guidelines (should start with a verb) + add task and PR ids in each commit message ?
+ write a quick description of what has been done in this PR + review PR title. 
(We will probably make this a rebase-ff)

File: addons/website_slides/controllers/main.py [hunk 3]
```diff
@@ -624,20 +624,24 @@ def slide_quiz_get(self, slide_id):
...
             } for question in slide.question_ids],
             'quizAttemptsCount': quiz_info['quiz_attempts_count'],
             'quizKarmaGain': quiz_info['quiz_karma_gain'],
             'quizKarmaWon': quiz_info['quiz_karma_won'],
         }
+        if slide.user_membership_id.sudo().completed:
+            for index_question, question in enumerate(slide.question_ids):
+                for index_answer, answer in enumerate(question.answer_ids):
+                    quiz_data['questions'][index_question]['answers'][index_answer]['is_correct'] = answer.is_correct
+        return quiz_data
```
Comment (dbeguin) [inline addons/website_slides/controllers/main.py]: I would rather add a variable to get once is slide is completed and use it directly in the returned dict: 
```
slide_is_completed = slide.user_membership_id.sudo().completed
quiz_data = {
            'questions': [{
                'id': question.id,
                'question': question.question,
                'answers': [{
                    'id': answer.id,
                    'text_value': answer.text_value,
                    'is_correct': answer.is_correct and slide_is_completed,
                } for answer in question.answer_ids],
            } for question in slide.question_ids],
            'quizAttemptsCount': quiz_info['quiz_attempts_count'],
            'quizKarmaGain': quiz_info['quiz_karma_gain'],
            'quizKarmaWon': quiz_info['quiz_karma_won'],
        }
```
But.. I don't really understand why this change ? what's the impact if we don't change this at all ?
Is this for rpc security information leak reasons ? -> If yes, write it in the code. Aaaaand, using 'my' method will return false for every answer. So that the hacker is screwed anyway :D

File: addons/website_slides/static/src/xml/slide_quiz.xml [hunk 4]
```diff
@@ -12,16 +12,15 @@
...
                                 <label class="mb-0 d-flex align-items-center justify-content-center mr-3 ml-3">
                                     <input type="radio"
                                         t-att-name="question.id"
                                         t-att-value="answer.id"
                                         class="d-none"/>
-                                    <i class="fa fa-circle text-muted"></i>
+                                    <i t-att-class="'fa fa-circle text-muted' + (!(widget.slide.completed &amp;&amp; answer.is_correct) ? '' : ' d-none')"></i>
                                     <i class="fa fa-times-circle text-danger d-none"></i>
-                                    <i class="fa fa-check-circle text-success d-none"></i>
+                                    <i t-att-class="'fa fa-check-circle text-success' + (widget.slide.completed &amp;&amp; answer.is_correct ? '' :  ' d-none')"></i>
```
Comment (dbeguin) [inline addons/website_slides/static/src/xml/slide_quiz.xml]: better to use %s then +

File: addons/website_slides/controllers/main.py (hunk 3 above)
Comment (qmo-odoo) [reply addons/website_slides/controllers/main.py]: JEM thought it was better to not have the key in the dict if the slide wasn't completed by the user. And yes, the change was made to prevent users to cheat the quiz.

File: addons/website_slides/static/src/xml/slide_quiz.xml (hunk 4 above)
Comment (dbeguin) [reply addons/website_slides/static/src/xml/slide_quiz.xml]: Ok my bad, didn't it was frontend template

File: addons/website_slides/controllers/main.py (hunk 3 above)
Comment (jem-odoo) [reply addons/website_slides/controllers/main.py]: try `None if not completed else answer.is_correct`
if None is transformed into undefined, then okay to do that.

Comment (dbeguin) [reply addons/website_slides/controllers/main.py]: Soooo, as discussed in a wonderful team brainstorming, we will go for 
`'is_correct': answer.is_correct if slide_is_completed else None.`
So that the variable is not returning any real value that could be misused by confusion
